source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,100 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+
+/**
+ * A hook for displaying link tooltips which can be used to visually
+ * enable users to click a link (without having to press down Ctrl/Cmd)
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (cm) => {
+  cm.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+
+    // Initialise displayed attributes to 'loading'
+    let title = 'loading'
+    let content = 'loading'
+    let wordCount = 'loading'
+    let time = 'loading'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false, // No arrow for these tooltips
+      onHidden (instance) {
+        instance.destroy() // Destroy the tippy instance.
```

Comment:
Actually, it might make sense not to destroy these tippys. I'm not sure about the performance impact of leaving the tippys, BUT I think this way we could save loading time if the user accidentally leaves the element and immediately hovers back over it (because if there's a `_tippy` associated with the link, then tippy will itself react to mousemove events and thus, re-display the tippy).

Would you mind taking a look into the Tippy documentation and maybe fiddle around a little bit with leaving the tippy vs. not leaving them? And also think about the potential usability impact if the notes get edited. If we leave the tippy instance and the user never scrolls, the loaded values might become outdated, the question would only be if that is ever a problem… I think not because as soon as the user switches documents (to edit the linked one) this `event.target` would get destroyed either way alongside the tippy and would be re-rendered either way.

So long story short, you may want to play around a little bit with some settings and tippy! You can also take a look at the formatting toolbar (it's either a hook or a plugin of the editor) which I implemented, since that one features some fancy tricks with the Tippy instance.

Reply:
It seems that leaving the tippy up does not prevent the content being changed. Probably because, as you said, when navigating away to change the tippy is removed anyway. Because of this, I have removed the onHidden event all together.

I think that this means that when a note is edited in a second of Zettlr, then the tooltip in the first instance won't be updated, but that doesn't seem like a massive deal compared to the cost of re-creating each hover time.